### ISSUE TITLE
Big issue with the delegate proxies

### ISSUE THREAD

**Issue by kevinrenskers** [CombineCommunity/CombineCocoa#31]
It seems that a delegate proxy can only be used on one item of its kind at a time. 

For example in this screen, there are two view controllers, both have a UISearchBar. The one in the main view controller works completely fine, but the one on the half-modal (that has focus) doesn't do anything, none of the new publishers (for example `searchBar.textDidChangePublisher`) fire. Until I remove the delegate publishers from the other search bar, and then it works in the modal.

<img width="478" alt="Screen Shot 2020-10-06 at 14 21 03" src="https://user-images.githubusercontent.com/229384/95200896-69db1900-07df-11eb-905d-6900673a3b46.png">

**Comment by freak4pc** [comment]
Definitely sounds like a bug. @jdisho let me know if you might have some time looking into this ? 
Otherwise I'll try carving some time in a couple of days.

**Comment by kevinrenskers** [comment]
Weirdly enough I can't reproduce the problem with a minimal repro project, but it's definitely a problem with my project. Not sure what's going on here. I'll investigate more and report back, hopefully with a repro project.

**Comment by freak4pc** [comment]
Any chance you're accidentally also setting a regular delegate with the proxy? Perhaps in a storyboard or something ?

**Comment by kevinrenskers** [comment]
No. But I did manage to boil it down to this:

``` swift
import UIKit
import Combine
import CombineCocoa

class SearchViewController: UIViewController {
  @IBOutlet var searchBar: UISearchBar!
  @IBOutlet var collectionView: UICollectionView!

  private var subscriptions = Set<AnyCancellable>()

  override func viewDidLoad() {
    super.viewDidLoad()

    searchBar.textDidChangePublisher
      .sink { print("Text changed: \($0)") }
      .store(in: &subscriptions)

    collectionView.didSelectItemPublisher
      .sink { print("Item selected: \($0)") }
      .store(in: &subscriptions)
  }
}
```

`searchBar.textDidChangePublisher` never fires, until you remove `collectionView.didSelectItemPublisher`.

**Comment by kevinrenskers** [comment]
Repro project: [CombineCocoaTest.zip](https://github.com/CombineCommunity/CombineCocoa/files/5334041/CombineCocoaTest.zip)

**Comment by jdisho** [comment]
Hey @freak4pc! Currently I'm in holidays and I will not be home until Sunday.

**Comment by freak4pc** [comment]
I'm pretty sure I know why this happens but will need to investigate.

**Comment by kevinrenskers** [comment]
I hope you can figure it out! Right now I have to rip out a whole bunch of delegate publishers from my app because yeah they're conflicting with each other. If there is anything I can do to help test a fix, let me know.

**Comment by freak4pc** [comment]
Fixed in 0.3.0. Thanks @TTOzzi !
